File: bronze/2501/c++/main.cpp
#include <iostream>

int main()
{

    int N, K;
    std::cin >> N >> K;
    int count = 0, result = 0;

    for (int i = 1; i <= N; i++) {
        if (N % i == 0) {
            count++;
        }
        if (count == K) {
            result = i;
            break;
        }
    }

    std::cout << result;

    return 0;
}

File: bronze/2501/python/main.py

if (__name__ == "__main__"):
    N, K = map(int, input().split())

    count, result = (0, 0)
    for i in range(1, N +1):
        if (N % i == 0): count += 1
        if (count == K): result = i; break
    print(result)